### ISSUE TITLE
VK_*_buffer_device_address missing deprecation note

### ISSUE THREAD

**Issue by krOoze** [KhronosGroup/Vulkan-Docs#1261]
`VK_KHR_buffer_device_address` and `VK_EXT_buffer_device_address` are missing deprecation note.

Per spec, deprecation requires explanation.

Furthermore `VK_KHR_buffer_device_address` is not marked as deprecated in xml.

**Comment by jeffbolznv** [comment]
Agreed. Would you mind proposing a PR?

**Comment by krOoze** [comment]
I do not know the reason for obsoletion. I only noticed the VU that prohibits to enable the extensions.

**Comment by oddhack** [comment]
@krOoze are you referring to this VU?

```
  * [[VUID-VkDeviceCreateInfo-pNext-04748]]
    if the pname:pNext chain includes a
    slink:VkPhysicalDeviceVulkan12Features structure and
    slink:VkPhysicalDeviceVulkan12Features::pname:bufferDeviceAddress is
    ename:VK_TRUE, pname:ppEnabledExtensionNames must: not contain
    `apiext:VK_EXT_buffer_device_address`
```

**Comment by krOoze** [comment]
@oddhack 

> [VUID-VkDeviceCreateInfo-ppEnabledExtensionNames-03328](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkDeviceCreateInfo-ppEnabledExtensionNames-03328)
`ppEnabledExtensionNames` **must** not contain both `VK_KHR_buffer_device_address` and `VK_EXT_buffer_device_address`

Which defacto means `VK_KHR_buffer_device_address` _obsoletes_ `VK_EXT_buffer_device_address` (i.e. the old cannot be used anymore if there is the new):

I vaguely remember the spec said obsoletion required justification to be stated in the spec, but can't find it anymore. Since it is relatively rare to obsolete something, should perhaps give explicit justification all the same.

Additionally the spec now says `VK_EXT_buffer_device_address` is deprecated, but per above it seems it is actually obsoleted in reality,

As for `VK_KHR_buffer_device_address` I am not sure. I suppose I interpreted "must not contain both" above as "must contain neither". The wording is bit unfortunate. I would  instead suggest "If contains `VK_KHR_buffer_device_address`, then it **must** not contain `VK_EXT_buffer_device_address`" wording.

**Comment by oddhack** [comment]
@baldurk would it be OK with you to forward your comments from the internal issue 2159 on why the exclusion VU exists? The current state is that we do have an intention to enforce the exclusion at the level of feature enable bits, not extension enables, but nobody has had time to develop that to the point of a PR, yet.

**Comment by baldurk** [comment]
Yes that's fine, to the best of my knowledge that's why the EXT extension was superceded and not compatible.

**Comment by oddhack** [comment]
Here's @baldurk's internal commentary:

> My memory is a little fuzzy but I think this was primarily done to allow validation to know what rules to apply. We aliased the buffer and memory bits between the EXT and KHR extensions, so if you enable EXT and KHR at the same time then validation loses its ability to highlight a specific problem: In the EXT extension you can bind a buffer with `VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT_EXT` to any old memory without special flags, because everything happens on the buffer. However if you bind a buffer with `VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT_KHR` to memory without `VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT` then that's an error. Because the bits are aliased then if both extensions are enabled it's impossible to know if you're in the KHR situation and you're in error, or if you're in the EXT situation and you're fine.
> 
> The reason I wanted that validation was indeed for capture/replay but it also applies to drivers that care about `VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT` directly like Intel's anv driver in mesa does. So for developers running on a driver with both exts enabled they don't get validation alerting them to the problem. I think in practice drivers that need the memory bit don't implement the EXT version, so the developer is unlikely to see a crash either.
> 
> I'm not sure how else to solve this - if you make the validation unconditionally/conservatively enforce the KHR rules then you're kind of retroactively saying that valid-EXT code is invalid as soon as the KHR extension is turned on, which could be really confusing. If you give up that validation then you have all the known risks that come with that.
